scripts/generate_upgrade_syncset.py

```diff
@@ -0,0 +1,301 @@
+#!/bin/env python
```

Comment:
I didn't check where/how this is going to be running, but do we want to be explicit about using python3? (Hint: We *really* [shouldn't be using python2](https://www.python.org/doc/sunset-python-2/).)

Reply:
👍  Python 2 is EOL.

Reply:
This will need to run on a bastion and the OSD3 Hive - sadly the latter only has python2. I've tested it in both py2 and py3; the /bin/env is picking up the correct python in both locations (that is to say, bastions will run it via python 3, OSD3 Hive will just call python 2). As this script has a limited life of 1-2 months I'm hoping it'll be a workable compromise.

Reply:
I had assumed you would be running it in a container in any case, is that not true?

Even then, I get that it doesn't seem worth the effort to change over. Unfortunately, taken individually, every other case of this is thinking the same way. So when we do finally decide to move, it'll be a death of a thousand cuts.

Not blocking on this, but I really want to put it in people's heads.